lib/scanner/walk.go

```diff
@@ -201,35 +207,58 @@ func (w *walker) walk(ctx context.Context) (chan protocol.FileInfo, error) {
 	return finishedChan, nil
 }
 
-func (w *walker) walkAndHashFiles(ctx context.Context, fchan, dchan chan protocol.FileInfo) fs.WalkFunc {
+func (w *walker) walkAndHashFiles(ctx context.Context, toHashChan, finishedChan chan ScanResult) fs.WalkFunc {
 	now := time.Now()
+	currDBFile, haveChanOpen := <-w.HaveChan
 	return func(path string, info fs.FileInfo, err error) error {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		default:
 		}
 
+		if path == "." {
+			return err
+		}
```

Comment:
I think this should be handled by fs.IsInternal()

Reply:
I don't think it does, it only checks for files matching `/.st**` as far as I see.

Reply:
Right, I'm thinking about my modified version in encryoted fs.